### ISSUE TITLE
[FEATURE] Request to add support for VPN client setup in Harvester node.

### ISSUE THREAD

**Issue by shan100github** [harvester/harvester#3312]
**Is your feature request related to a problem? Please describe.**
While we install Harvester OS it becomes difficult to manage servers when the hardware is deployed remotely.
When we have a custom VPN like Wireguard/Other, we can remotely monitor and manage the Harvester node.

I would like to request support setup of the `Wireguard client or another VPN client` as a part of the Harvester node installation/configuration.

**Additional context**
Referring: https://github.com/harvester/harvester/issues/2476

**Comment by xeruf** [comment]
Yes we need that as well, somehow the nodes need to be able to phone to the outside as well!

**Comment by shan100github** [comment]
Feature not necessary anymore

**Comment by xeruf** [comment]
Huh? I just said I want it too? Should I open a new identical issue?
What is your solution?
If you don't want to be notified about it anymore, simply reopen the issue and disable notifications for yourself.

See also https://slack-archive.rancher.com/t/8214812/is-it-possible-to-have-wireguard-vpn-setup-in-the-harvester-

**Comment by w13915984028** [comment]
@xeruf We may re-open this issue, and add your detailed requirements as well.

We would like to know more details about the scenario.

**Comment by shan100github** [comment]
FYI: we have added a router with wireguard VPN along with the baremetal node at present.

**Comment by w13915984028** [comment]
@shan100github Could you add more details of your setup:

In a Harvester cluster, there is a cluster level `VIP` which is deployed to one of the management `NODEs` dynamically; and each `NODE` has an unique `NODE` IP, also used as `NODE`'s identity.

When the VPN client is sitting in an Router which is out of the Harvester cluster, then the Harvester cluster has no knowledge of this VPN.

When the VPN client is setting inside Harvester cluster, putting the `VPN client` behind `VIP` or `NODE IP` is the first question. Then it comes to the question of injecting/withdrawing VPN routes to the `NODE` (dynamically).

**Comment by xeruf** [comment]
We have the nodes sitting behind a router running wireguard, and by becoming wireguard clients, they could have securely routed internet access.

**Comment by shan100github** [comment]
@w13915984028 
we need the wireguard vpn running because physical servers will be placed in remote locations and need to secure way to manage the physical servers.
So we have decided to add a hardware VPN router along with the remotely placed physical server & create a VPN mesh network. (Since we are using hardware VPN router, it helped  managing physical server as well rather than at OS level)

if we had a VPN supported on harvester we could have avoided hardware VPN router. But for us, we decided to go on with hardware VPN router for this project.
